kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -879,6 +879,37 @@ class ResidualBasedNewtonRaphsonStrategy
         KRATOS_CATCH("");
     }
 
+    /**
+     * @brief Gets the current solution vector
+     * @param rDofSet The set of degrees of freedom (DOFs)
+     * @return A vector containing the current solution values for the given DOF set
+     * @details This method retrieves the current solution values for the provided DOF set. Each value is accessed using the equation ID associated with each DOF.
+     */
+    Vector GetCurrentSolution(DofsArrayType& rDofSet){
+        Vector this_solution = ZeroVector(rDofSet.size());
+        for (auto& r_dof : rDofSet) {
+            this_solution[r_dof.EquationId()] = r_dof.GetSolutionStepValue();
+        }
+        return this_solution;
+    }
+
+    /**
+     * @brief Gets the matrix of non-converged solutions and the DOF set
+     * @return A tuple containing the matrix of non-converged solutions and the DOF set
+     * @details This method returns a tuple where the first element is a matrix of non-converged solutions and the second element is the corresponding DOF set.
+     */
+    std::tuple<Matrix, DofsArrayType> GetNonconvergedSolutions(){
+        return std::make_tuple(mNonconveregedSolutionsMatrix, GetBuilderAndSolver()->GetDofSet());
+    }
+
+    /**
+     * @brief Sets up the gathering of non-converged solutions
+     * @details This method enables the storage of non-converged solutions at each iteration by setting the appropriate flag.
+     */
+    void SetUpNonconvergedSolutionsGathering(){
+        mStoreNonconvergedSolutionsFlag = true;
```

Comment:
It is a good practice to make the member var match the name of the inquiry and set methods.

Reply:
Method's name now matches the flag's